docs/topic-guides/swift/suggest.md

```diff
@@ -0,0 +1,109 @@
+
+```swift
+import class MozillaAppServices.SuggestStore
+import class MozillaAppServices.SuggestStoreBuilder
+import class MozillaAppServices.Viaduct
+import enum MozillaAppServices.SuggestionProvider
+import enum MozillaAppServices.RemoteSettingsServer
+import struct MozillaAppServices.SuggestIngestionConstraints
+import struct MozillaAppServices.SuggestionQuery
+```
+
+On start up of your application, you create a `SuggestStore` (as a singleton). You do this via the `SuggestStoreBuilder`, which returns a `SuggestStore`:
+
+```swift
+let store: SuggestStore
+
+var builder = SuggestStoreBuilder()
+    .dataPath(path: dataPath)
+
+if let remoteSettingsServer {
+    builder = builder.remoteSettingsServer(server: remoteSettingsServer)
+}
+
+store = try builder.build()
+```
+
+In addition to the `dataPath`, which is the path (the SQLite location) where you store your suggestions, you need a `remoteSettingsServer`, which is the URL to the instance of `RemoteSettings` you will be using (e.g. `https://remote-settings-dev.allizom.org`)
+
+## Ingesting suggestions
+
+We ensure that the Rust networking stack has been initialized via:
+
+```Swift
+import class MozillaAppServices.Viaduct
+
+Viaduct.shared.useReqwestBackend()
+```
+
+This is safe to call multiple times.
+
+### On Start Up
+Ingest with `SuggestIngestionConstraints::empty_only=true` shortly after each startup. This ensures we have something in the DB on the first run and also after upgrades where we often will clear the DB to start from scratch.
+
+### Periodically
+
+Ingest with `SuggestIngestionConstraints::empty_only=false` on regular schedule (like once a day).
+
+Example:
+
+```swift
+try self.store.ingest(constraints: SuggestIngestionConstraints(emptyOnly: true))
+```
+
+The defintion of `SuggestIngestionConstraints`:
+
+```swift
+public struct SuggestIngestionConstraints {
+    public var providers: [SuggestionProvider]?
+    public var providerConstraints: SuggestionProviderConstraints?
+    /**
+     * Only run ingestion if the table `suggestions` is empty
+     */
+    public var emptyOnly: Bool
+
+    // Default memberwise initializers are never public by default, so we
+    // declare one manually.
+    public init(providers: [SuggestionProvider]? = nil, providerConstraints: SuggestionProviderConstraints? = nil, 
+        /**
+         * Only run ingestion if the table `suggestions` is empty
+         */emptyOnly: Bool = false) {
+        self.providers = providers
+        self.providerConstraints = providerConstraints
+        self.emptyOnly = emptyOnly
+    }
+}
+```
+
+## Querying Suggestions
+
+Call `SuggestStore::query` to fetch suggestions for the suggest bar:
+
+```swift
+try self.store.query(query: SuggestionQuery(
+    keyword: keyword,
+    providers: providers,
+    limit: limit
+)).compactMap(RustFirefoxSuggestion.init)
```

Comment:
I don't think the `compactMap` is required it's something firefox-ios does in order to use its own class for suggestions.  I'm not sure if this is something we should recommend or not so maybe leave it out for now.